### PR TITLE
Add doctest continuous integration

### PR DESCRIPTION
**Pull request by Strilanc**
Fixes https://github.com/quantumlib/Cirq/issues/1816

### CODE REVIEW

File: cirq/linalg/decompositions_perf_test.py [hunk 1]
```diff
@@ -2,7 +2,19 @@
...
+# yapf: disable
+SWAP = np.array([[1, 0, 0, 0],
+                 [0, 0, 1, 0],
+                 [0, 1, 0, 0],
+                 [0, 0, 0, 1]])
+CNOT = np.array([[1, 0, 0, 0],
+                 [0, 1, 0, 0],
+                 [0, 0, 0, 1],
+                 [0, 0, 1, 0]])
+CZ = np.diag([1, 1, 1, -1])
```
Comment (cduck) [inline cirq/linalg/decompositions_perf_test.py]: `# yapf: enable`

File: check/doctest [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Usage:
+#     check/doctest
+################################################################################
+
+# Get the working directory to the repo root.
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$(git rev-parse --show-toplevel)"
+
+python_source_files=(`find cirq | grep "\.py$"`)
+python -m doctest "${python_source_files[@]}"
```
Comment (cduck) [inline check/doctest]: Putting all the files on the command line risks exceeding the maximum command length.  Also user arguments aren't passed to doctest.  Just use find: `find cirq -type f -iname '*.py' | xargs python -m doctest $@`.

File: cirq/linalg/decompositions_perf_test.py (hunk 1 above)
Comment (Strilanc) [reply cirq/linalg/decompositions_perf_test.py]: Done.

File: check/doctest (hunk 2 above)
Comment (Strilanc) [reply check/doctest]: Not having user arguments was on purpose (note it's not mentioned in the usage instructions). Used your line without the $@.

File: cirq/linalg/decompositions_perf_test.py [hunk 3]
```diff
@@ -30,3 +30,4 @@
 def test_kak_decomposition_perf(target, benchmark):
     kak = benchmark(cirq.kak_decomposition, target)
     np.testing.assert_allclose(cirq.unitary(kak), target, atol=1e-8)
+# yapf: enable
```
Comment (cduck) [inline cirq/linalg/decompositions_perf_test.py]: Do you need yapf disabled from lines 19 to 32?

Comment (Strilanc) [reply cirq/linalg/decompositions_perf_test.py]: Nope. Fixed.

**Pull request merged**